example/public/index.html

```diff
@@ -67,65 +66,63 @@ <h1>Display BPMN 2.0 Diagrams Using React</h1>
       console.log('diagram shown');
     }
 
-    class ExternalLoadedBpmn extends React.Component {
-
-      constructor(props) {
-        super(props);
-
-        this.state = {};
-
-        const { url } = props;
-
-        fetch(url).then(res => res.text()).then(diagramXML => {
-          this.setState({
-            diagramXML
-          });
-        });
-      }
-
-      render() {
-
-        const {
-          onError,
-          onShown,
-          onLoading
-        } = this.props;
-
-        const { diagramXML } = this.state;
-
-        return (
-          diagramXML
-            ? <ReactBpmn
-                diagramXML={ diagramXML }
-                onLoading={ onLoading }
-                onShown={ onShown }
-                onError={ onError }
-              />
-            : null
-        );
-      }
-
+    function ExternalLoadedBpmn({ url }) {
+      const [diagramXML, setDiagramXML] = React.useState()
+
+      React.useEffect(() => {
+        const abortController = new AbortController();
+        const loadXml = async () => {
+          try {
+            const response = await fetch(url, {
+              signal: abortController.signal
+            })
+            const result = await response.text()
+            setDiagramXML(result)
+
+          } catch (error) {
+            if (!abortController.signal.aborted) {
+              console.log(error);
+            }
+          }
+        }
+
+        loadXml()
+
+        return () => abortController.abort();
+      }, [url])
+
+      return (
+        diagramXML
+          ? <ReactBpmn
+            diagramXML={diagramXML}
+            onLoading={onLoading}
+            onShown={onShown}
+            onError={onError}
+          />
+          : null
+      );
     }
 
-    ReactDOM.render(
+    const container = document.querySelector('#diagram');
+    const root = ReactDOM.createRoot(container)
```

Comment:
Is `ReactDOM.render` reprecated? If not I'd rather use it as it si the post simple version.

Reply:
`ReactDOM.render` is deprecated in React 18: https://react.dev/blog/2022/03/29/react-v18#react-dom-client

Reply:
Thanks, did not know! :)